src/scripts/pages/home.js

```diff
@@ -23,15 +22,19 @@ document.addEventListener('DOMContentLoaded', () => {
   }
 
   function showPublicCategories () {
-    $publicCategoriesSection.classList.add('active');
-    $privateCategoriesSection.classList.remove('active');
-    $toggleBtn.innerText = privateBtn;
+    if ($publicCategoriesSection && $privateCategoriesSection && $toggleBtn) {
+      $publicCategoriesSection.classList.add('active');
+      $privateCategoriesSection.classList.remove('active');
+      $toggleBtn.innerText = privateBtn;
+    }
   }
 
   function showPrivateCategories () {
-    $publicCategoriesSection.classList.remove('active');
-    $privateCategoriesSection.classList.add('active');
-    $toggleBtn.innerText = publicBtn;
+    if ($publicCategoriesSection && $privateCategoriesSection && $toggleBtn) {
+      $publicCategoriesSection.classList.remove('active');
+      $privateCategoriesSection.classList.add('active');
+      $toggleBtn.innerText = publicBtn;
+    }
   }
 
   function showVisibleCategory () {
```

Comment:
We'll need a check on line 68 `$toggleBtn.addEventListener('click', toggleCategories);`. You could also use the optional chaining operator (`?.`)

So either:
```js
$toggleBtn?.addEventListener('click', toggleCategories);
```

or:
```js
if ($toggleBtn) {
    $toggleBtn.addEventListener('click', toggleCategories);
}
```